### DISCUSSION TITLE
/opt/homebrew/Library/Homebrew/shims/super/xcrun (Errno::ENOENT)

### DISCUSSION THREAD

**Discussion by scanterog** [Homebrew/discussions#3602] [Everyday usage]
Hello!

We have a private formula that runs a `bazelisk build` command to generate some static C library (.a file). This is failing with

```
19mn0vw/.brew_home/_bazel/b8314d0cb53a12b831b8600014154a77/external/rust_darwin_aarch64/lib/rustlib/aarch64-apple-darwin/lib/libcompiler_builtins-78b101bb38333bc9.rlib" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/private/tmp/libstreaming-20220830-50275-19mn0vw/.brew_home/_bazel/b8314d0cb53a12b831b8600014154a77/external/rust_darwin_aarch64/lib/rustlib/aarch64-apple-darwin/lib" "-o" "bazel-out/darwin_arm64-opt-exec-312AC71C/bin/external/rules_rust/util/process_wrapper/process_wrapper" "-Wl,-dead_strip" "-nodefaultlibs" "-Wl,-no-as-needed" "-Wl,-z,relro,-z,now" "-undefined" "dynamic_lookup" "-headerpad_max_install_names" "-lSystem" "-lresolv" "-lc++" "-lm"
  = note: /opt/homebrew/Library/Homebrew/shims/super/cc:489:in `exec': No such file or directory - /opt/homebrew/Library/Homebrew/shims/super/xcrun (Errno::ENOENT)
          	from /opt/homebrew/Library/Homebrew/shims/super/cc:489:in `<main>'
```

It works if I create a symlink for `/opt/homebrew/Library/Homebrew/shims/super/xcrun` to `/usr/bin/xcrun`. As you can see this is aarch64 but it's also happening for x86.

I'm not sure where the problem comes from:
* https://github.com/bazelbuild/rules_rust/tree/main/util/process_wrapper trying to use the wrong xcrun path? I also see there's a xcrun here https://github.com/Homebrew/brew/blob/master/Library/Homebrew/shims/mac/super/xcrun.
* I've read that the xcode developer tool should provide this and it does apparently but I don't fully understand how this works.
```
xcrun --show-sdk-path
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
```

Any pointers? Thanks!

**Comment by hanshuebner** [comment]
@scanterog Have you ever been able to solve this?  We're running into the same problem after having switched to Bazel.  Thanks!

**Comment by carlocab** [comment]
Bazel likes to reset the environment, and that doesn't interact well with Homebrew's build environment (which uses environment variables to configure the behaviour of the shims).

Adding
```ruby
  env :std
```
to your formula might help.

**Reply by hanshuebner** [reply]
Adding `env :std` to our formula indeed helped quite a lot and prevents me from having to explicitly weed out the environment variables required by Homebrew.  Thanks much!
